### DISCUSSION TITLE
Do SourceGenerators currently need to be netstandard2.0?

### DISCUSSION THREAD

**Discussion by msedi** [dotnet/roslyn#51640] [Q&A] [answered]
I wanted to try our SourceGenerators and followed the projects in VS 16.9.0 (official version - no preview). Both, my testapp and the SourceGenerator were `<TargetFramework>.net5.0</TargetFramework>` but it didn't work. When checking the few examples I found that the SourceGenerator is marked with `netstandard2.0`. 

Is this currently a requirement or am I doing something wrong. If this is necessary it would be nice if it would be documented somewhere.

**Comment by jmarolf** [comment, marked as answer]
All analyzers and generators must target netstandard2.0

**Reply by msedi** [reply]
@jmarolf: Thanks for the info. I'm curious, is this documented somewhere? And what is the reason for this?

**Reply by sharwell** [reply]
> And what is the reason for this?

The tools run under both .NET Framework 4.7.2 (IDE scenarios, `msbuild`) and .NET Core (`dotnet build`, `dotnet msbuild`). **netstandard2.0** is the newest framework which ensures both scenarios will work.

**Reply by msedi** [reply]
@sharwell. That makes sense. Thanks for the info.
